File: src/ingestion/src/domain/documents.rs
//! Mendefinisikan dokumen, struktur pasal/ayat, identitas ketentuan, dan versi teks beserta metadata waktu.
//!
//! Peran dalam komponen:
//! Menjadi kontrak ingestion, penyimpanan metadata, graph, dan filter temporal.
//!
//! Kontrak integrasi dan perhatian implementasi:
//! Bedakan regulation ID, provision ID, provision-version ID, source artifact ID, tanggal berlaku, dan tanggal observasi; status unknown eksplisit.
//!
//! Benchmark dan gate penerimaan:
//! [DOMAIN] Gate deterministik: ID dan referensi sumber/versi tidak ambigu; invalid record ditolak atau dikarantina secara eksplisit. Kompatibilitas schema diperiksa dengan fixtures; belum ada hasil pengukuran.
//!
//! [VERSIONING] Ukur ketepatan pemilihan versi pada pertanyaan bertanggal, kebocoran versi yang tidak berlaku, dan penanganan status tidak diketahui. Gate: tanggal observasi tidak menggantikan tanggal berlaku; fixtures perubahan/pencabutan/unknown menghasilkan status yang ditentukan label.
//! Target numerik required: configs/benchmark-targets.yaml; status REQUIRED_UNMEASURED.
//! Target hanya boleh diubah dengan persetujuan pengguna; ikuti doc/benchmark-policy.md.
//!
//! Status: scaffold dokumentasi; perilaku modul belum diimplementasikan.
//!
//! Kontrak wire bersama berasal dari src/contracts. Tipe Rust nantinya hanya representasi lokal; field wire tidak didefinisikan ulang secara independen.
//! Rekomendasi implementasi berikutnya (belum merupakan fitur aktif):
//! Construct validated document/source/provision views over generated types; keep observation time separate from legal dates.
//! Bukti verifikasi: Test stable IDs, raw/normalized mappings and historical version ambiguity; avoid redundant conversion/allocation across batches.
//! Target numerik tetap configs/benchmark-targets.yaml; ikuti doc/verification.md.

File: src/server/internal/api/schemas/documents.go
// Mendefinisikan request ingestion dan response status dokumen/job.
//
// Peran dalam komponen:
// Memisahkan penerimaan dokumen dari pekerjaan pemrosesan yang panjang.
//
// Kontrak integrasi dan perhatian implementasi:
// Validasi ukuran/format dan identitas job; status accepted tidak berarti indeks sudah tersedia untuk retrieval.
//
// Benchmark dan gate penerimaan:
// [API] Gate: input/response mengikuti schema, error tidak disamarkan menjadi jawaban sukses, dan readiness sesuai dependency yang diperlukan. Ukur latency end-to-end p50/p95, error rate, concurrency, dan timeout pada workload yang dinyatakan; target benchmark wajib ada di configs/benchmark-targets.yaml; hasil belum diukur.
//
// [UPDATE] Gate: sumber tak berubah tidak diekstraksi ulang; update identik idempotent; dependensi terdampak diinvalidasi; bukti yang masih didukung sumber lain tetap tersedia. Uji pemulihan kegagalan parsial dan snapshot konsisten. Ukur waktu/biaya update terhadap full rebuild serta freshness lag.
// Target numerik required: configs/benchmark-targets.yaml; status REQUIRED_UNMEASURED.
// Target hanya boleh diubah dengan persetujuan pengguna; ikuti doc/benchmark-policy.md.
//
// Status: scaffold dokumentasi; perilaku modul belum diimplementasikan.
// Rekomendasi implementasi berikutnya (belum merupakan fitur aktif):
// Map document/version/job payloads without leaking SDK objects or storage paths.
// Bukti verifikasi: Test historical versions, paging cursors and failed/partial job responses using contract fixtures.
// Target numerik tetap configs/benchmark-targets.yaml; ikuti doc/verification.md.

package schemas
